src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexGeneratorAttribute.cs

```diff
@@ -0,0 +1,44 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Threading;
+
+namespace System.Text.RegularExpressions;
+
+/// <summary>Instructs the System.Text.RegularExpressions source generator to generate an implementation of the specified regular expression.</summary>
+[AttributeUsage(AttributeTargets.Method, AllowMultiple = false, Inherited = false)]
```

Comment:
Can it happen that a developer defines another version of the attribute that have `AllowMultiple = true` or different `AttributeTargets`, which can be returned by `GetTypeByMetadataName`?

I have no idea if this can happen, so just noting to consider.

Reply:
> Can it happen that a developer defines another version of the attribute that have AllowMultiple = true or different AttributeTargets, which can be returned by GetTypeByMetadataName?

In theory, sure. MultipleRegexGeneratorAttributes should flag multiple.  If someone creates their own and then uses it to attribute other things, that's on them; I don't think we should be significantly increasing the cost or complexity of the generator/analyzer for such things.